src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueCollateralAdaptor.sol

```diff
@@ -0,0 +1,222 @@
+    // error MorphoBlueCollateralAdaptor__MarketPositionsMustBeTracked(Id id);
+
+    /**
+     * @notice Attempted to interact with an Morpho Blue Lending Market the Cellar is not using.
+     */
+    error MorphoBlueCollateralAdaptor__MarketPositionsMustBeTracked(MarketParams market);
+
+    /**
+     * @notice Removal of collateral causes Cellar Health Factor below what is required
+     */
+    error MorphoBlueCollateralAdaptor__HealthFactorTooLow(MarketParams market);
+
+    /**
+     * @notice Minimum Health Factor enforced after every removeCollateral() strategist function call.
+     * @notice Overwrites strategist set minimums if they are lower.
+     */
+    uint256 public immutable minimumHealthFactor;
+
+    /**
+     * @param _morphoBlue immutable Morpho Blue contract (called `Morpho.sol` within Morpho Blue repo).
+     * @param _healthFactor Minimum Health Factor that replaces minimumHealthFactor. If using new _healthFactor, it must be greater than minimumHealthFactor. See `BaseAdaptor.sol`.
+     */
+    constructor(address _morphoBlue, uint256 _healthFactor) MorphoBlueHelperLogic(_morphoBlue) {
+        _verifyConstructorMinimumHealthFactor(_healthFactor);
+        morphoBlue = IMorpho(_morphoBlue);
+        minimumHealthFactor = _healthFactor;
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Blue Collateral Adaptor V 0.1"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice User deposits collateralToken to Morpho Blue market.
+     * @param assets the amount of assets to provide as collateral on Morpho Blue.
+     * @param adaptorData adaptor data containing the abi encoded Morpho Blue market.
+     * @dev configurationData is NOT used.
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        // Deposit assets to Morpho Blue.
+        MarketParams memory market = abi.decode(adaptorData, (MarketParams));
+        _validateMBMarket(market);
+        ERC20 collateralToken = ERC20(market.collateralToken);
+        _addCollateral(market, assets, collateralToken);
+    }
+
+    /**
+     * @notice User withdraws are NOT allowed from this position.
+     * NOTE: collateral withdrawal calls directly from users disallowed for now.
+     */
+    function withdraw(uint256, address, bytes memory, bytes memory) public pure override {
+        revert BaseAdaptor__UserWithdrawsNotAllowed();
```

Comment:
The user can deposit using `deposit` or `addCollateral`, but can only withdraw using `removeCollateral`. Is it intended? Why?

Reply:
So user deposits will go through the adaptors `deposit` function. ie When a user deposits into the vault, the money they deposited can be directly moved to morpho blue in the same TX. 

Where as `addCollateral` is how a strategist rebalancing the vault would add more collateral to morpho blue. 

We do not support user withdraws from morpho blue collateral bc withdraws will lower the health factor of the cellar, so we would need to calculate the health factor and make sure it is not below some minimum value. We could do it, but in practice we have found it much more gas efficient for users to just wait for the strategist to rebalance collateral out of the lending protocol first.